src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -377,7 +377,8 @@ public int Run()
                         logger, typeSystemContext, XmlReader.Create(fs), substitutionFilePath, featureSwitches));
             }
 
-            ilProvider = new FeatureSwitchManager(ilProvider, logger, featureSwitches, substitutions);
+            SubstitutionProvider substitutionProvider = new SubstitutionProvider(logger, featureSwitches, substitutions);
+            ilProvider = new SubstitutedILProvider(ilProvider, substitutionProvider);
```

Comment:
Do we also need to fix a similar piece of code in the tests: https://github.com/dotnet/runtime/blob/0f70e9aa5f08847680162a6dc712e069f0334b52/src/coreclr/tools/aot/ILCompiler.Trimming.Tests/TestCasesRunner/TrimmingDriver.cs#L114

Reply:
Ah yes. "runtime (Build linux-x64 checked CLR_Tools_Tests) Failing after 11m" seems to agree